### PR TITLE
Add OMArk and create data manager OMamer

### PR DESCRIPTION
**Pull request by rlibouba**
OMark is a software requiring OMamer databases.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by bgruening** [comment]
Thanks for starting this work!

File: tools/omark/omark.xml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    #end if
+
+    #if $verbose:
+        -v
+    #end if
+
+    ]]></command>
+
+    <inputs>
+        <param name="input" type="data" format="fasta" label="Input fasta file" help="Each gene is represented by a single protein sequence"/>
```
Comment (abretaud) [inline tools/omark/omark.xml:35]:
```suggestion
        <param name="input" type="data" format="fasta" label="Protein sequences" help="Each gene is represented by a single protein sequence"/>
```

File: tools/omark/omark.xml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        <param name="input_iso" type="data" format="txt" optional="true" label="Input isoform file" help="A semi-colon separated file, listing all isoforoms of each genes, with one gene per line."/>
```
Comment (abretaud) [inline tools/omark/omark.xml:36]:
```suggestion
        <param name="input_iso" type="data" format="txt" optional="true" label="Input isoform file" help="A semi-colon separated file, listing all isoforms of each genes, with one gene per line."/>
```

Isn't this contradictory with the previous option help "Each gene is represented by a single protein sequence" ?

File: tools/omark/omark.xml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        <param argument="-r" type="text" optional="true" label="Taxonomic rank">
+            <sanitizer invalid_char="">
+                <valid initial="string.letters,string.digits">
+                    <add value="_" />
+                </valid>
+            </sanitizer>
+            <validator type="regex">[0-9a-zA-Z_]+</validator>
+        </param>
+
+        <param name="og_fasta" type="boolean"  checked="false" label="Using the fasta file for analysis"/>
```
Comment (abretaud) [inline tools/omark/omark.xml:60]: Not clear what it means

File: tools/omark/omark.xml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        <param name="verbose" type="boolean"  checked="false" label="Turn on logging information about OMArk process"/>
```
Comment (abretaud) [inline tools/omark/omark.xml:61]: I don't think we want to expose this option, just add it or not by default in the cmd

File: tools/omark/omark.xml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    <inputs>
+        <param name="input" type="data" format="fasta" label="Input fasta file" help="Each gene is represented by a single protein sequence"/>
+        <param name="input_iso" type="data" format="txt" optional="true" label="Input isoform file" help="A semi-colon separated file, listing all isoforoms of each genes, with one gene per line."/>
+        <param argument="-c" name="omark_mode" type="boolean"  checked="false" label="Switch OMArk mode to only computing a list of conserved HOGs and output it as list" help="Can be used to obtain a set of genes on which to train"/>
+
+        <param name="database" label="OMAmer database" type="select">
+            <options from_data_table="omamer"/>
+        </param>
+
+        <param argument="-t" type="text" optional="true" label="Taxonomic identifier">
```
Comment (abretaud) [inline tools/omark/omark.xml:43]: Not clear what these taxonomic options are used for

File: data_managers/data_manager_omamer/data_manager/omamer.xml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        <requirement type="package" version="2.0.2">omamer</requirement>
+    </requirements>
+    <version_command></version_command>
+    <command detect_errors="exit_code"><![CDATA[
+        python '$__tool_directory__/omamer.py'
+        --name '$name'
+        --json '$output_file'
+    ]]></command>
+    <inputs>
+        <param name="name" type="text" label="Unique ID" value="" />
```
Comment (abretaud) [inline data_managers/data_manager_omamer/data_manager/omamer.xml:12]: A select box would be better

File: data_managers/data_manager_omamer/data_manager/omamer.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+    # Download each specified OMAmer data set
+    for dataset in OMAMER_DATASETS:
+        url = OMAMER_DATASETS_URL.format(dataset=dataset)
+        base_name = os.path.splitext(dataset)[0]
+        destination_path = os.path.join(args.output_dir, base_name)
+        download_file(url, destination_path)
+
+    # Utiliser le nom du fichier sans extension comme identifiant unique
+    data_manager_entry = {
```
Comment (abretaud) [inline data_managers/data_manager_omamer/data_manager/omamer.py:52]: version is missing here

**Review by abretaud — COMMENTED** [review]
A few comments

**Comment by rlibouba** [comment]
Hello ! 
I'm having the same problem as with the compleasm PR (https://github.com/galaxyproject/tools-iuc/pull/5664). I want to import the data manager's database list into the wrapper's xml.

**Comment by bgruening** [comment]
@rlibouba let us know when this one is ready for review :)

**Comment by rlibouba** [comment]
Hi, it's ready for review :)

File: tools/omark/omark.xml [hunk 8]
```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0"?>
+<tool id="omark" name="OMArk" version="@TOOL_VERSION@+galaxy0" profile="21.05">
+    <description></description>
```
Comment (bgruening) [inline tools/omark/omark.xml:3]: Can you please add a description here what this tool is doing?

Comment (bgruening) [reply tools/omark/omark.xml:3]:
```suggestion
    <description>proteome quality assessment</description>
```

**Review by bgruening — APPROVED** [review]

**Pull request merged**